### PR TITLE
ERR: Raise NotImplementedError with BaseIndexer and certain rolling operations

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref #32865
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

As a temporary solution for returning incorrect results when using a BaseIndexer subclass with certain rolling operations, a `NotImplementedError` will now raise with instructions to use `apply` instead.

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -1173,7 +1181,11 @@ class _Rolling_and_Expanding(_Rolling):
     )
 
     def count(self):
-
+        if isinstance(self.window, BaseIndexer):
```
Comment (jreback) [inline pandas/core/window/rolling.py]: can you make this a common method, e.g.
```self.validate_support_of_indexer()``

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/rolling.py [hunk 2]
```diff
@@ -145,6 +145,20 @@ def _validate_get_window_bounds_signature(window: BaseIndexer) -> None:
                 f"get_window_bounds"
             )
 
+    @staticmethod
+    def _validate_baseindexer_support(window, func_name: Optional[str]) -> None:
```
Comment (jreback) [inline pandas/core/window/rolling.py]: either make this a method or just a module level free function ; i find static methods are
non obvious to grok

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/rolling.py (hunk 2 above)
Comment (mroeschke) [reply pandas/core/window/rolling.py]: Done. Created a common function instead of a static method

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**